app/src/main/java/com/douglasbruce/tasky/BootCompletedReceiver.kt

```diff
@@ -0,0 +1,28 @@
+package com.douglasbruce.tasky
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import com.douglasbruce.tasky.core.domain.utils.AlarmScheduler
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class BootCompletedReceiver : BroadcastReceiver() {
+
+    @Inject
+    lateinit var alarmScheduler: AlarmScheduler
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        if (intent?.action == Intent.ACTION_BOOT_COMPLETED) {
+            val scope = CoroutineScope(Job())
+            scope.launch(Dispatchers.IO) {
+                alarmScheduler.scheduleAllFutureAlarms()
+            }
```

Comment:
There's no difference between this coroutine scope and `GlobalScope`, just that you can cancel this custom scope which you don't do here.
Custom scopes make sense if you are building a component with a custom lifecycle. In this case, I'd just use `GlobalScope`